cfme/tests/services/test_request.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+from cfme.services.catalogs.catalog_item import CatalogItem
+from cfme.automate.service_dialogs import ServiceDialog
+from cfme.services.catalogs.catalog import Catalog
+from cfme.services.catalogs.service_catalogs import ServiceCatalogs
+from cfme.services import requests
+from cfme.web_ui import flash
+from utils import testgen
+from utils.providers import setup_provider
+from utils.randomness import generate_random_string
+from utils.log import logger
+from utils.wait import wait_for
+from utils import version
+import utils.randomness as rand
+
+pytestmark = [
+    pytest.mark.meta(server_roles="+automate"),
+    pytest.mark.usefixtures('logged_in', 'vm_name', 'uses_infra_providers'),
+    pytest.mark.long_running,
+    pytest.mark.ignore_stream("5.2")
+]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.infra_providers(metafunc, 'provisioning')
+
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+        if not args['provisioning']:
+            # No provisioning data available
+            continue
+
+        # required keys should be a subset of the dict keys set
+        if not {'template', 'host', 'datastore'}.issubset(args['provisioning'].viewkeys()):
+            # Need all three for template provisioning
+            continue
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append(argvalues[i])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
+
+
+@pytest.fixture()
+def provider_init(provider_key):
```

Comment:
Almost missed this...

`provider_init` should go away, use [setup_provider](https://github.com/RedHatQE/cfme_tests/blob/master/fixtures/provider.py#L43) instead.